pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py

```diff
@@ -91,14 +91,14 @@ def _VSCODE_convertTensorToDataFrame(tensor, start=None, end=None):
 def _VSCODE_convertToDataFrame(df, start=None, end=None):
     vartype = type(df)
     if isinstance(df, list):
-        df = _VSCODE_pd.DataFrame(df)
+        df = _VSCODE_pd.DataFrame(df).iloc[start:end]
     elif isinstance(df, _VSCODE_pd.Series):
-        df = _VSCODE_pd.Series.to_frame(df)
+        df = _VSCODE_pd.Series.to_frame(df).iloc[start:end]
     elif isinstance(df, dict):
         df = _VSCODE_pd.Series(df)
-        df = _VSCODE_pd.Series.to_frame(df)
+        df = _VSCODE_pd.Series.to_frame(df).iloc[start:end]
     elif hasattr(df, "toPandas"):
-        df = df.toPandas()
+        df = df.toPandas().iloc[start:end]
```

Comment:
Broke chunking in https://github.com/microsoft/vscode-jupyter/pull/4951/

Reply:
Sorry, but the comment doesn't make sense.
It looks like this is a bug fix.

Reply:
I'm a tiny bit confused as well. This is merging into more-styles and currently there is also a PR open for more-styles into main?

Reply:
Yeah this PR depends on the Checkbox change in more-styles. But I'm merging the style changes into main, then this into main. And yes this is a bug fix.

Reply:
Ok, that works for me. I think that personally I would just park it until style was into main and then PR against main so I could see test runs and what not. But small enough change that I'm down with approving now.

Reply:
Somehow I thought we used to get test runs even on PRs not going into main... That's totally fair, my bad. Can't wait for GitHub to add stacked diffs 😭